### PR TITLE
start of Appearance function

### PR DESCRIPTION
**Pull request by jessfraz**
- hook up https://github.com/replit/Codemirror-CSS-color-picker/blob/master/src/index.ts (will do in another pr because we will have to fork this for our syntax)

### CODE REVIEW

File: src/wasm-lib/kcl/src/std/appearance.rs [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+    // Validate the data.
+    data.validate().map_err(|err| {
+        KclError::Semantic(KclErrorDetails {
+            message: format!("Invalid appearance data: {}", err),
+            source_ranges: vec![args.source_range],
+        })
+    })?;
+
+    // Make sure the color if set is valid.
+    let re = Regex::new(r"^#[0-9a-fA-F]{6}$").unwrap();
```
Comment (jtran) [inline src/wasm-lib/kcl/src/std/appearance.rs:49]: Can we not recompile this every call? The readme has an example using `static`: https://github.com/rust-lang/regex#usage-avoid-compiling-the-same-regex-in-a-loop

File: src/wasm-lib/kcl/src/std/appearance.rs [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+    let result = inner_appearance(data, solid_set, args).await?;
+    Ok(result.into())
+}
+
+/// Set the appearance of a solid. This only works on solids, not sketches or individual paths.
+///
+/// This will work on any solid, including extruded solids, revolved solids, and shelled solids.
+/// ```no_run
+/// /// Add color to an extruded solid.
+/// const exampleSketch = startSketchOn("XZ")
```
Comment (jtran) [inline src/wasm-lib/kcl/src/std/appearance.rs:66]: We should stop using `const`. They're warnings now.

```suggestion
/// exampleSketch = startSketchOn("XZ")
```

**Review by jtran — APPROVED** [review]
This is great! 🤩

File: src/wasm-lib/kcl/src/std/appearance.rs (hunk 1 above)
Comment (jessfraz) [reply src/wasm-lib/kcl/src/std/appearance.rs:49]: good catch using lazy_static now

File: src/wasm-lib/kcl/src/std/appearance.rs (hunk 2 above)
Comment (jessfraz) [reply src/wasm-lib/kcl/src/std/appearance.rs:66]: on it sorry i copy postad from other files

**Pull request merged**